xbmc/music/Album.cpp

```diff
@@ -56,6 +57,33 @@ CAlbum::CAlbum(const CFileItem& item)
 
   if (!tag.GetMusicBrainzAlbumArtistID().empty())
   { // have musicbrainz artist info, so use it
+    // Establish tag consistency - do the number of musicbrainz ids and number of names in hints or artist match
+    if (tag.GetMusicBrainzAlbumArtistID().size() != musicBrainAlbumArtistHints.size() &&
+      tag.GetMusicBrainzAlbumArtistID().size() != tag.GetAlbumArtist().size())
+    {
+      // Tags mis-match - report it and then try to fix
+      CLog::Log(LOGDEBUG, "Mis-match in song file albumartist tags: %i mbid %i name album: %s",
+        tag.GetMusicBrainzAlbumArtistID().size(), tag.GetAlbumArtist().size(), strAlbum);
+      /*
+      Most likey we have no hints and a single artist name like "Artist1 feat. Artist2"
+      or "Composer; Conductor, Orchestra, Soloist" or "Artist1/Artist2" where the
+      expected single item separator (default = space-slash-space) as not been used.
+      Comma and slash (no spaces) are poor delimiters as could be in name e.g. AC/DC,
+      but here treat them as such in attempt to find artist names.
+      */
+      if (tag.GetAlbumArtist().size() == 1)
```

Comment:
I was avoiding trying to fix all kinds of possible mess, and just trying to fix the known formats that make up 99% of the problems users have. I have learnt with tag data that if try to get too clever it bites you!

I would say do this limited repair and see what user problems are left afterwards. But thanks for thinking about it.